### ISSUE TITLE
False negative for negate-equal-op (SIM201)

### ISSUE THREAD

**Issue by last-partizan** [astral-sh/ruff#9106]
I'm using ruff 0.1.7, and in some cases it does not detect SIM201 case.

Take a look at this:

```python
value = 0

if not value == 0:  # Incorrect: No error
    raise ValueError("Invalid data")

if not value == 0:  # Correct: SIM201 [*] Use `value != 0` instead of `not value == 0`
    pass
```

```
> ruff --isolated --select SIM201 /tmp/test.py

/tmp/test.py:6:4: SIM201 [*] Use `value != 0` instead of `not value == 0`
Found 1 error.
```

But there should be two errors.

**Comment by last-partizan** [comment]
Oh, that's desired behavior.

Found this https://github.com/astral-sh/ruff/issues/6684, closing.
